livre/core/pipeline/Executable.h

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2011-2014, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _Executable_h_
+#define _Executable_h_
+
+#include <livre/core/types.h>
+#include <livre/core/pipeline/Future.h>
+
+namespace livre
+{
+
+/**
+ * The Executable class is the base class for objects that can be executed by workers.
+ */
+class Executable
```

Comment:
Just a comment.

Reading this interface alone without more context than this is a pipeline library points out that the naming could be improved, but I don't have a concrete suggestion to improve.
A class called Executable shouldn't provide much more than an execute method. I understand why getOutFutures and getConnectedInFutures are there, but for a casual reader it's not obvious what these are about and why they are needed.

Do you think getPreconditions ang getPostconditions are better names?
I think that with those names, the return type and no additional context, it's more or less clear that they are the futures you have to wait for to know that you can execute the object and that it finished.